### PR TITLE
Adding Linux and Mac CI runs.

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @joshfree, @shiftylogic, @mellinoe 

Borrowed snippets from https://github.com/dotnet/corefx/blob/master/netci.groovy#L366-L441

### CODE REVIEW

**Comment by ahsonkhan** [comment]
This diff is easier to see (extra spacing makes it harder to see what changed):
image

**Comment by shiftylogic** [comment]
Is it possible to test this without merging it?

**Comment by joshfree** [comment]
This doesn't look like it'll work.  It appears to call the windows build.cmd on linux and Mac as-is.

**Comment by ahsonkhan** [comment]
> This doesn't look like it'll work. It appears to call the windows build.cmd on linux and Mac as-is.


Yes, I am waiting for this PR to get merged first: https://github.com/dotnet/corefxlab/pull/1532

**Comment by ahsonkhan** [comment]
> Is it possible to test this without merging it?

I am trying to figure that out with the help of @crummel

**Comment by ahsonkhan** [comment]
cc @crummel, please review :)

File: netci.groovy [hunk 1]
```diff
@@ -8,36 +8,46 @@ def branch = GithubBranchName
...
-        // and calls made within that closure apply to the newly created job.
-        def newJob = job(newJobName) {
+            // Create a new job with the specified name.  The brace opens a new closure
+            // and calls made within that closure apply to the newly created job.
+            def newJob = job(newJobName) {
+                
+                // This opens the set of build steps that will be run.
+                steps {
+                    // Indicates that a batch script should be run with the build string (see above)
+                    // Also available is:
```
Comment (crummel) [inline netci.groovy]: Comment is a little awkward now that we're using "shell" as well.  Could move explanation of "shell" to above the shell command.

**Comment by mellinoe** [comment]
You should be able to test this through the regular Jenkins system, although I'm not sure if it's hooked up in corefxlab.

@dotnet-bot Test CI please

**Comment by mellinoe** [comment]
https://ci.dot.net/job/dotnet_corefxlab/job/master/job/GenPRTest/job/osx10.12_debug_prtest
https://ci.dot.net/job/dotnet_corefxlab/job/master/job/GenPRTest/job/ubuntu16.04_debug_prtest/

You can test those out when https://github.com/dotnet/corefxlab/pull/1532 is merged.

**Comment by ahsonkhan** [comment]
@mellinoe, #1532 is merged. How can we test those?

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
@dotnet-bot Test CI please

**Comment by ahsonkhan** [comment]
@dotnet-bot Test CI please

**Pull request merged**